okio/src/main/java/okio/Buffer.java

```diff
@@ -1253,6 +1253,28 @@ Segment writableSegment(int minimumCapacity) {
     return -1L;
   }
 
+  @Override public long indexOf(ByteString bytes) throws IOException {
+    return indexOf(bytes, 0);
+  }
+
+  @Override public long indexOf(ByteString bytes, long fromIndex) throws IOException {
+    int bytesSize = bytes.size();
+    loop:
+    while ((fromIndex = indexOf(bytes.getByte(0), fromIndex)) != -1) {
+      if (!request(fromIndex + bytesSize)) {
+        return -1; // Not enough bytes to match.
+      }
+      for (int i = 1; i < bytesSize; i++) {
```

Comment:
Turns out we have this on `ByteString` for matching others called `rangeEquals`.